src/bin/pg_autoctl/pghba.c

```diff
@@ -624,3 +648,60 @@ pghba_enable_lan_cidr(PGSQL *pgsql,
 	}
 	return true;
 }
+
+
+/*
+ * hba_check_hostname returns true when the DNS setting looks compatible with
+ * Postgres expectations for an HBA hostname entry.
+ *
+ * https://www.postgresql.org/docs/current/auth-pg-hba-conf.html
+ *
+ * If a host name is specified (anything that is not an IP address range or a
+ * special key word is treated as a host name), that name is compared with the
+ * result of a reverse name resolution of the client's IP address (e.g.,
+ * reverse DNS lookup, if DNS is used). Host name comparisons are case
+ * insensitive. If there is a match, then a forward name resolution (e.g.,
+ * forward DNS lookup) is performed on the host name to check whether any of
+ * the addresses it resolves to are equal to the client's IP address. If both
+ * directions match, then the entry is considered to match. (The host name that
+ * is used in pg_hba.conf should be the one that address-to-name resolution of
+ * the client's IP address returns, otherwise the line won't be matched. Some
+ * host name databases allow associating an IP address with multiple host
+ * names, but the operating system will only return one host name when asked to
+ * resolve an IP address.)
+ */
+bool
+pghba_check_hostname(const char *hostname)
+{
+	char ipaddr[BUFSIZE] = { 0 };
+
+	/*
+	 * IP addresses do not require any DNS properties/lookups. Also hostname
+	 * won't contain a '/' character, but CIDR notations would, such as
+	 * 1.2.3.4/32 or ::1/128. We don't want to trust ip_address_type() value of
+	 * IPTYPE_NONE when we find a '/' character in the hostname.
+	 *
+	 */
+	if (strchr(hostname, '/') || ip_address_type(hostname) != IPTYPE_NONE)
+	{
+		return true;
+	}
+
+	if (!resolveHostnameForwardAndReverse(hostname, ipaddr, sizeof(ipaddr)))
+	{
+		/* warn users about possible DNS misconfiguration */
+		log_warn("Failed to resolve hostname \"%s\" to an IP address that "
+				 "resolves batck to the hostname on a reverse DNS lookup.",
+				 hostname);
+
+		log_warn("Postgres might deny connection attempts from \"%s\", "
+				 "even with the new HBA rules.",
```

Comment:
I think it's good to add some more actionable info in a log message on how to fix this issue, e.g.
> You probably don't want to use hostnames in your HBA config, but instead use IPs, since IPs are much easier to set up correctly. If you do want hostnames, you should set up the DNS records correctly as described in the postgres docs <link to postgres docs here>.

Reply:
Added an hint in that vein, though not this exact phrase.